modules/Core/pages/panel/registration.php

```diff
@@ -109,18 +65,79 @@
                     $configuration->set('Core', 'recaptcha_type', Input::get('captcha_type'));
                 }
 
-                // reCAPTCHA key
-                $configuration->set('Core', 'recaptcha_key', Input::get('recaptcha'));
+                // Verify if the captha inputted is correct (if enabled)
+                if (Input::get('enable_recaptcha') == 1 || Input::get('enable_recaptcha_login') == 1) {
+                    if (
+                        (
+                            CaptchaBase::isCaptchaEnabled('recaptcha_login')
+                            || CaptchaBase::isCaptchaEnabled()
+                        )
+                        && (
+                            CaptchaBase::getActiveProvider()->validateSecret(Input::get('recaptcha_secret')) == false
+                            || CaptchaBase::getActiveProvider()->validateKey(Input::get('recaptcha')) == false
+                        )
+                    ) {
+                        $errors = [$language->get('user', 'invalid_recaptcha')];
```

Comment:
Hmm, maybe an error specific to "hey person you didnt enter the right info" is better than a generic one?